web/api/queries/query.c

```diff
@@ -1093,6 +1218,22 @@ static RRDR *rrd2rrdr_fixedstep(
             r->internal.grouping_add   = grouping_add_average;
             r->internal.grouping_flush = grouping_flush_average;
         }
+
+        //assign statistic functions
+        j = 0;
+        for(i = 0; api_v1_data_stats[i].name; i++) {
```

Comment:
A suggestion, instead to declare `j = 0`, and to do the incremental inside the for loop, you can write:

` for(i = 0, j = 0; api_v1_data_stats[i].name; i++, j++) {`